src/loader/bgw_launcher.c

```diff
@@ -76,8 +96,13 @@ typedef struct DbHashEntry
 	Oid			db_oid;			/* key for the hash table, must be first */
 	BackgroundWorkerHandle *db_scheduler_handle;	/* needed to shut down
 													 * properly */
+	SchedulerState state;
+	VirtualTransactionId vxid;
+	int			state_advance_failures;
```

Comment:
I first interpreted this variable as "failures happening in advance" and I couldn't make sense of it. In other words, I think this naming might be confusing, because I now believe you mean to refer to state transition failures?

Reply:
renamed to state_transition_failures